packages/data-context/src/data/ProjectConfigIpc.ts

```diff
@@ -107,6 +111,16 @@ export class ProjectConfigIpc extends EventEmitter {
     return super.emit(evt, ...args)
   }
 
+  hasTypeScriptInstalled () {
+    try {
+      require.resolve('typescript', { paths: [this.projectRoot] })
+
+      return true
+    } catch (e) {
+      return false
+    }
+  }
+
```

Comment:
Can we share this function with the above file so we just have one implementation of it?

Reply:
Done.